src/tracing/trace_flamegraph.c

```diff
@@ -0,0 +1,270 @@
+/*
+   +----------------------------------------------------------------------+
+   | Xdebug                                                               |
+   +----------------------------------------------------------------------+
+   | Copyright (c) 2002-2020 Derick Rethans                               |
+   +----------------------------------------------------------------------+
+   | This source file is subject to version 1.01 of the Xdebug license,   |
+   | that is bundled with this package in the file LICENSE, and is        |
+   | available at through the world-wide-web at                           |
+   | https://xdebug.org/license.php                                       |
+   | If you did not receive a copy of the Xdebug license and are unable   |
+   | to obtain it through the world-wide-web, please send a note to       |
+   | derick@xdebug.org so we can mail you a copy immediately.             |
+   +----------------------------------------------------------------------+
+ */
+#include "php.h"
+#include "ext/standard/php_string.h"
+
+#include "php_xdebug.h"
+#include "tracing_private.h"
+#include "trace_flamegraph.h"
+
+#include "lib/lib_private.h"
+#include "lib/var_export_line.h"
+
+extern ZEND_DECLARE_MODULE_GLOBALS(xdebug);
+
+flamegraph_list *list_item_create()
+{
+	flamegraph_list *ret;
+
+	ret = xdmalloc(sizeof(flamegraph_list));
+	ret->head = NULL;
+
+	return ret;
+}
+
+void list_item_free(flamegraph_list *values)
+{
+	flamegraph_list_item *cur, *prev;
+
+	cur = values->head;
+	while (cur) {
+		prev = cur->next;
+		xdfree(cur);
+		cur = prev;
+	}
+	xdfree(values);
+}
+
+/* Increment or set value at given key, values are added to head since we will
+   always attempt to access recently added items, when processing direct
+   children in execution stack. */
+void list_item_inc(flamegraph_list *values, int key, int inc)
```

Comment:
I was thinking about making `list_item_inc()` being `static inline` to enforce faster calls, considering that tracing is very slow to being with, and this will be called hundred of thousands times, but as a C beginner, I don't master the consequences, I'd leave this decision to you. 

Same reasoning `list_item_del()` as well.

Reply:
And also for `compute_inclusive_value()`.

Reply:
I did it, seems much cleaner than my previous solution.